### DISCUSSION TITLE
SSO Contracts - Create2 for Account Deployment

### DISCUSSION THREAD

**Discussion by 4NNNN** [zkSync-Community-Hub/zksync-developers#850] [SSO]
[Account Implementation](https://github.com/matter-labs/zksync-sso-clave-contracts/blob/8a14c35a0b46333cb3dbdca83606ea0bc522faee/scripts/deploy.ts#L81) can't be deployed using normal create2. If that's the case, the account would be treated as a normal contract rather than an create2Account deployed "Account".

This wasn't possible before because if the account is a normal contract, any call to the proxy of the account would end up in "touched disallowed storage slots revert". Did you guys go for a testnet simulation and how is this possible now?

**Comment by 4NNNN** [comment]
> [Account Implementation](https://github.com/matter-labs/zksync-sso-clave-contracts/blob/8a14c35a0b46333cb3dbdca83606ea0bc522faee/scripts/deploy.ts#L81) can't be deployed using normal create2. If that's the case, the account would be treated as a normal contract rather than an create2Account deployed "Account".
> 
> This wasn't possible before because if the account is a normal contract, any call to the proxy of the account would end up in "Untouched storage slots revert". Did you guys go for a testnet simulation and how is this possible now?

[639](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/639)

**Reply by kaymomin** [reply]
cc @MexicanAce

**Reply by JackHamer09** [reply]
@4NNNN I think all SSO contracts are deployed using `create2`. We indeed faced `touched disallowed storage slots` error and had to whitelist specific sso contracts on the server level to prevent that.
Not sure I understand rest of the question. Let me know if this answers it.

**Reply by 4NNNN** [reply]
@JackHamer09 The question context revolves around [SSOAccount](https://github.com/matter-labs/zksync-sso-clave-contracts/blob/main/src/SsoAccount.sol) deployment. I wanted to know, how did you guys achieve to bypass "untouched storage slot revert" and what exactly was causing it!

Server whitelisting is the last call - What made you guys whitelist few contracts of the SSO codebase and why couldn't you guys figure out the storage slot reverts. It would be nice if you give us specifics, because we're working on it as well!

**Reply by JackHamer09** [reply]
I'll try to get more details about that and get back to you

**Reply by JackHamer09** [reply]
Hey @4NNNN 
Whitelisting is needed so that accounts can read their implementation address from the beacon during validation.
This happens because of some very strict security rules, but I think this should change in future protocol versions.
You can also read more about this here: https://docs.zksync.io/zksync-protocol/account-abstraction/extending-4337#extending-the-set-of-slots-that-belong-to-a-user

**Reply by 4NNNN** [reply]
@JackHamer09 Figured, I just replaced beacon proxy implementation with a 1967 proxy implementation.

**Reply by XinyuCRO** [reply]
Hi @JackHamer09 , seems that adding SessionKeyValidator, WebAuthValidator, SsoBeacon addresses to the paymaster access whitelist is not enough, can I know the full address list when deploying the whole zksync-sso setup to a zkstack chain?
